File: android/app/src/main/kotlin/com/usesmileid/sampleapps/android/navigation/UseSmileIDSampleDeepLinks.kt
package com.usesmileid.sampleapps.android.navigation

/**
 * The deep-link URI for every route. Paths are shared across the four sample apps; the scheme is
 * per app, which is why these live in the shell and never in the shared UI module.
 */
internal object UseSmileIDSampleDeepLinks {
    const val SCHEME = "usesmileid-sample-android"

    const val PRODUCTS = "$SCHEME://products"
    const val VERIFICATIONS = "$SCHEME://verifications"
    const val VERIFICATION_DETAILS = "$SCHEME://verifications/{jobId}"
    const val SETTINGS = "$SCHEME://settings"

    const val CONSENT_DETAILS_FORM = "$SCHEME://flow/{productId}/details"
    const val ID_DETAILS_FORM = "$SCHEME://flow/{productId}/id-details"
    const val COUNTRY_PICKER = "$SCHEME://flow/{productId}/id-details/country"
    const val ID_TYPE_PICKER = "$SCHEME://flow/{productId}/id-details/id-type"
    const val SDK_FLOW = "$SCHEME://flow/{productId}/run?route={route}"

    const val PROFILES = "$SCHEME://profiles"
    // androidx ranks argument-free deep links first, so these beat /profiles/{profileId}.
    const val PROFILE_SWITCH = "$SCHEME://profiles/switch"
    const val NEW_PROFILE = "$SCHEME://profiles/new"
    const val PROFILE_CONFIG = "$SCHEME://profiles/{profileId}"

    const val SCAN_TOKEN = "$SCHEME://token/scan"
    const val SCENARIO_DRAWER = "$SCHEME://debug/scenarios"

    // A dev-only shell surface, so it is not in spec/routes.json. It still needs a link, because
    // reaching it cold is the only way a flow can drive it.
    const val COMPONENT_GALLERY = "$SCHEME://debug/components"
}

File: ios/App/Sources/Navigation/UseSmileIDSampleDeepLinks.swift
/// The deep-link URI per route: paths are shared across the four apps, the scheme is per app, so these live in the shell.
enum UseSmileIDSampleDeepLinks {
  static let scheme = "usesmileid-sample-ios"

  static let products = "\(scheme)://products"
  static let verifications = "\(scheme)://verifications"
  static let verificationDetails = "\(scheme)://verifications/{jobId}"
  static let settings = "\(scheme)://settings"

  static let consentDetailsForm = "\(scheme)://flow/{productId}/details"
  static let idDetailsForm = "\(scheme)://flow/{productId}/id-details"
  static let countryPicker = "\(scheme)://flow/{productId}/id-details/country"
  static let idTypePicker = "\(scheme)://flow/{productId}/id-details/id-type"
  static let sdkFlow = "\(scheme)://flow/{productId}/run?route={route}"

  static let profiles = "\(scheme)://profiles"
  static let profileSwitch = "\(scheme)://profiles/switch"
  static let newProfile = "\(scheme)://profiles/new"
  static let profileConfig = "\(scheme)://profiles/{profileId}"

  static let licenses = "\(scheme)://settings/licenses"

  static let scanToken = "\(scheme)://token/scan"
  static let scenarioDrawer = "\(scheme)://debug/scenarios"
}
